### PR TITLE
Updated nix completer with the new flake APIs

### PR DESCRIPTION
**Pull request by aftix**
- Created an ActionInstallables for installables from flakes or nixpkgs
- Added `--stdin` flags to commands taking installables
- Added several subcommands under `nix flake`
- Removed old subcommands that have been renamed or removed from the nix program

Due to https://github.com/carapace-sh/carapace-bin/issues/2348#issuecomment-2059487194 tab completion for flake outputs was not implemented. Also, `nix help` follows the somewhat common pattern of putting subcommands as positional arguments such as `nix help flake show` to show help for `nix flake show`. I wasn't aware of a good programatic way to generate those completions.

### CODE REVIEW

**Comment by rsteube** [comment]
Regarding the help command:
```go
carapace.Gen(helpCmd).PositionalAnyCompletion(
    carapace.ActionCommands(rootCmd),
)
```
e.g. https://github.com/carapace-sh/carapace-bin/blob/e95f07dba3c876d947784b5de530fc1597ce5c24/completers/dagger_completer/cmd/help.go#L20

**Comment by aftix** [comment]
I've added the ActionCommands to `nix help` and rebased against master again

**Review by rsteube — CHANGES_REQUESTED** [review]

**Comment by aftix** [comment]
I've made the requested changes, but wasn't exactly sure what kind of error messages would be good on failure

File: pkg/actions/tools/nix/installables.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package nix
+
+import (
+	"strings"
+
+	"github.com/carapace-sh/carapace"
+)
+
```
[the remark this answers is not in the document]
Comment (rsteube) [reply pkg/actions/tools/nix/installables.go:8]: Ah, sorry :sweat_smile: . The oneliner would have been enough (it's used for completion and the macro site in the docs).

**Comment by rsteube** [comment]
Just `err.Error()` is fine.

**Comment by aftix** [comment]
I've updated the ActionMessage calls to use `err.Error()`

**Review by rsteube — APPROVED** [review]

**Comment by rsteube** [comment]
thanks

**Pull request merged**